
extends Control

var dialogue = [
#<<<<<<< HEAD
#=======
		"They permit smart men to live and useful men to work.", 
		"As we are neither, there lie ahead only two paths:",
		"We may die, or we may fight, and I, for one, am too cowardly to die.",
		". . . Here is my ultimatum.",
		"We gather here today as traitors, as enemies of the state.",
		"By setting foot in this room, you have put yourselves on the path of no return.",
		"Know that there are only two sides to this conflict. Betray us, and you will have nowhere to turn.",
		"That said, I will give you one opportunity to leave. Stand, and be forgotten. Stay, and be condemned.",
		". . . Good. Now we may begin.",
		"The bombs have been placed under the E.L.F factory, and I have the detonator.",
		"Good riddance. However, remember the factories are not our only objective. ",
		"To make sure nothing like this ever happens again, we . . .",
		"are going to have to do some serious spring cleaning.",
		
#>>>>>>> f0599f3fa4c9156509b07ab3f949457bf47c4d97
]
var dialogue_index = 0
var finished = false
var interact = false

func _ready():
	$Sprite.hide()
	yield(get_tree().create_timer(1), "timeout")
	load_dialogue()

func _process(_delta):
	$"continuesprite".visible = finished
	if Input.is_action_just_pressed("ui_accept") and finished == true:
		load_dialogue()
		
func load_dialogue():
	$Sprite.show()
	yield(get_tree().create_timer(0.5), "timeout")
	if dialogue_index < dialogue.size():

		$Type.play()
		finished = false
		$RichTextLabel.bbcode_text = dialogue[dialogue_index]
		$RichTextLabel.percent_visible = 0
		$Tween.interpolate_property(
			$RichTextLabel, "percent_visible", 0, 1, 2, 
			Tween.TRANS_LINEAR, Tween.EASE_IN_OUT
		)
		$Tween.start()
	
	else:
		queue_free()
	
func _on_Tween_tween_completed(_object, _key):
	finished = true
	$Type.stop()
	dialogue_index +=1



func _on_Control_tree_exited():
	Transition.change_scene("res://act3/jailscenes/jail4.tscn")
